### PR TITLE
runqueue: use newtypes for ids

### PR DESCRIPTION
**Pull request by elenaf9**
Use Newtypes for `RunqueueId` and `ThreadId`, instead of type aliases. 
This prevents mistakes, especially when a method like `Runque::add()` takes both, a `RunqueueId` and a `ThreadId` as parameter.
See also the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/type-safety.html#newtypes-provide-static-distinctions-c-newtype).

### CODE REVIEW

File: src/riot-rs-runqueue/src/runqueue.rs
Comment (elenaf9) [reply src/riot-rs-runqueue/src/runqueue.rs]: Done in e77a204.

File: src/riot-rs-runqueue/src/runqueue.rs
Comment (elenaf9) [reply src/riot-rs-runqueue/src/runqueue.rs]: > Would it make sense to make the inner value private instead, and provide a new() constructor, so that the value cannot be modified after the instance has been created?

Done in e77a204.

> Ideally, these newtypes would only be created in this module/crate, so that we could make the constructor private/pub(crate)-private, not sure whether this is possible/makes sense however.

That's unfortunately not possible. `riot-rs-threads` creates `RunqueueId`s based on the set thread priority, and `ThreadId`s are created based on availability. The logic for both is part of `riot-rs-threads`.

Comment (elenaf9) [reply src/riot-rs-runqueue/src/runqueue.rs]: Yes, I thought about that as well. I was leaning towards leaving it as it is, because `riot-rs-runqueue` doesn't really know anything about thread priorities and only cares about runqueues.
But then again, if we'd want to be consistent and have the naming in `riot-rs-runqueue` completely abstract, it should also be called `ItemId` instead of `ThreadId`.

So yes, I am in favor of renaming it to `ThreadPriority`. cc @kaspar030  wdyt?

Comment (kaspar030) [reply src/riot-rs-runqueue/src/runqueue.rs]: fine with me.
maybe split out into a different PR. your call.

Comment (elenaf9) [reply src/riot-rs-runqueue/src/runqueue.rs]: :+1:. Will do a different PR for it after this is merged.

**Comment by elenaf9** [comment]
The hax workflow currently fails with

```sh
 Error 308:
  - Recursive dependency on module Riot_rs_runqueue.Runqueue.Clist.fst
```

I am not that familiar with hax. Is this something that I can/ have to fix in our implementation or a bug in hax?

**Comment by elenaf9** [comment]
> The hax workflow currently fails with
> 
> ```shell
>  Error 308:
>   - Recursive dependency on module Riot_rs_runqueue.Runqueue.Clist.fst
> ```
> 
> I am not that familiar with hax. Is this something that I can/ have to fix in our implementation or a bug in hax?

Solved by using `u8`s instead of the newtypes in the Clist submodule. Not ideal, but since its only internal I think it's acceptable.

**Review by kaspar030 — APPROVED** [review]

**Pull request merged**
